src/compile.ts

```diff
@@ -58,14 +58,20 @@ export function compile(
         if (isAppsyncResolver(node)) {
           return visitAppsyncResolver(node);
         } else if (isReflectFunction(node)) {
-          return toFunction("FunctionDecl", node.arguments[0]);
+          const func = node.arguments[0];
+          if (ts.isIdentifier(func)) {
+            throw new Error(
+              `Reflect does not support function references: ${func.text}`
```

Comment:
Should we move this logic to the `toFunction` call since the same restrictions applies to all cases.